lib/web/gziphandler.go

```diff
@@ -0,0 +1,59 @@
+package web
+
+import (
+	"compress/gzip"
+	"io"
+	"net/http"
+	"strings"
+)
+
+type gzipResponseWriter struct {
+	io.Writer
+	http.ResponseWriter
+}
+
+// Write uses the Writer part of gzipResponseWriter to write the output.
+func (w gzipResponseWriter) Write(b []byte) (int, error) {
+	// Explicitly set Content-Type, otherwise the browser gets confused.
+	if "" == w.Header().Get("Content-Type") {
+		// If no content type, apply sniffing algorithm to un-gzipped body.
+		w.Header().Set("Content-Type", http.DetectContentType(b))
+	}
```

Comment:
could you remind what is this one needed? For example, we have headers which explicitly tell the browsers not to do the content sniffing as the browsers might be tricked to do things they are not supposed to.

Reply:
resolved in slack: [content sniffing is skipped](https://github.com/golang/go/issues/31753) if content-encoding header is set, so we need sniff it ourselves before writing the compressed data